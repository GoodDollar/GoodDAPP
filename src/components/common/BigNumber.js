// @flow
import React from 'react'
import { View, StyleSheet } from 'react-native'
import { Text } from 'react-native-paper'
import normalize from 'react-native-elements/src/helpers/normalizeText'
import { fontStyle } from './styles'

/**
 * Receives a number and a unit to display
 * @param {Props} props
 * @param {Number} [props.number] Number to show
 * @param {String} [props.unit] Units for the number
 * @param {Object} [props.elementStyles] Inner elements styles
 * @param {Object} [props.style] Outer element style
 * @returns {React.Node}
 */
const BigNumber = (props: any) => (
  <View style={[styles.bigNumberWrapper, props.style]}>
    <Text style={[styles.bigNumber, props.elementStyles]}>{props.number}</Text>
    <Text style={[styles.bigNumberUnit, props.elementStyles]}>{props.unit}</Text>
  </View>
)

const styles = StyleSheet.create({
  bigNumberWrapper: {
    display: 'inline-block'
  },
  bigNumber: {
    ...fontStyle,
    fontSize: normalize(30),
    textAlign: 'right'
  },
  bigNumberUnit: {
    ...fontStyle,
    textAlign: 'right'
  }
})

export default BigNumber
