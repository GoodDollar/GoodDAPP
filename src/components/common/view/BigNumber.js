// @flow
import React from 'react'
import { View } from 'react-native'
import { withStyles } from '../../../lib/styles'
import { theme } from '../../theme/styles'
import Text from './Text'

/**
 * Receives a number and a unit to display
 * @param {Props} props
 * @param {Number} [props.number] Number to show
 * @param {String} [props.unit] Units for the number
 * @param {Object} [props.elementStyles] Inner elements styles
 * @param {Object} [props.style] Outer element style
 * @returns {React.Node}
 */
class BigNumber extends React.Component {
  render() {
    const {
      bigNumberStyles,
      bigNumberUnitStyles,
      bigNumberProps,
      bigNumberUnitProps,
      children,
      number,
      unit,
      style,
      color,
      styles,
      testID,
      reverse,
      spaceBetween = true,
    } = this.props

    const components = []

    if (reverse) {
      components.push(
        unit ? (
          <Text
            key="big_number_unit"
            fontFamily={theme.fonts.slab}
            fontSize={18}
            lineHeight={18}
            fontWeight="bold"
            textAlign="right"
            color={color || 'gray'}
            {...bigNumberUnitProps}
            style={bigNumberUnitStyles}
          >
            {unit}
          </Text>
        ) : (
          children
        ),
        spaceBetween ? <Text key="big_number_space"> </Text> : null,
        <Text
          key="big_number_amount"
          fontFamily={theme.fonts.slab}
          fontSize={36}
          lineHeight={36}
          fontWeight="bold"
          textAlign="right"
          color={color || 'gray'}
          {...bigNumberProps}
          style={[styles.bigNumber, bigNumberStyles]}
        >
          {number}
        </Text>,
      )
    } else {
      components.push(
        <Text
          key="big_number_amount"
          fontFamily={theme.fonts.slab}
          fontSize={36}
          fontWeight="bold"
          textAlign="right"
          color={color || 'gray'}
          lineHeight={36}
          {...bigNumberProps}
          style={[styles.bigNumber, bigNumberStyles]}
        >
          {number}
        </Text>,
        unit ? (
          <Text
            key="big_number_unit"
            fontFamily={theme.fonts.slab}
            fontSize={18}
            lineHeight={18}
            fontWeight="bold"
            textAlign="right"
            color={color || 'gray'}
            {...bigNumberUnitProps}
            style={bigNumberUnitStyles}
          >
            {unit}
          </Text>
        ) : (
          children
        ),
      )
    }

    return (
      <View style={[styles.bigNumberWrapper, style]} testID={testID}>
        {components}
      </View>
    )
  }
}

const getStylesFromProps = ({ theme }) => {
  return {
    bigNumberWrapper: {
      alignItems: 'center',
      display: 'flex',
      flexDirection: 'row',
    },
    bigNumber: {
      marginRight: 2,
    },
  }
}

export default withStyles(getStylesFromProps)(BigNumber)
