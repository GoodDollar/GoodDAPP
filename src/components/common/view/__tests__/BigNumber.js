import React from 'react'
import renderer from 'react-test-renderer'
import BigNumber from '../BigNumber'
import { withThemeProvider } from '../../../../__tests__/__util__'

// Note: test renderer must be required after react-native.

describe('BigNumber', () => {
  const WrappedBigNumber = withThemeProvider(BigNumber)

  it('matches snapshot', async () => {
    let component

    // eslint-disable-next-line require-await
    await renderer.act(async () => (component = renderer.create(<WrappedBigNumber number={10} unit="G$" />)))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
