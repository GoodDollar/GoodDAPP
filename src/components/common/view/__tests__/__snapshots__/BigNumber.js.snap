// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigNumber matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flexDirection-18u37iz"
  >
    <div
      className="css-text-901oao r-color-181bzza r-fontFamily-6jnb14 r-fontSize-hiw28u r-fontWeight-b88u0q r-lineHeight-qvk6io r-marginRight-a5pmau r-textAlign-1ff274t r-textDecorationLine-13wfysu r-textTransform-3twk1y"
      dir="auto"
    >
      10
    </div>
    <div
      className="css-text-901oao r-color-181bzza r-fontFamily-6jnb14 r-fontSize-hiw28u r-fontWeight-b88u0q r-lineHeight-qvk6io r-textAlign-1ff274t r-textDecorationLine-13wfysu r-textTransform-3twk1y"
      dir="auto"
    >
      G$
    </div>
  </div>
</div>
`;
